ls1intum/Artemis
# `Communication`: Load conversations in course overview only when needed

### Checklist
#### General


#### Client

### Motivation and Context


Improve the performance of Artemis by delaying the loading of conversations until the user actually navigates to them.

### Description

Currently Artemis loads all messages when entering a course. But since students often just want to participate in exercises, this is not necessary, so the call to load all messages is moved to the messages tab itself.
Instead a small REST call is made to determine if there are unread messages and a red dot should be displayed next to the message tab.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Course with messages enabled

1. Log in to Artemis
2. Go into the course overview and inspect the networks tab
3. Make sure that there was no `conversations` call
4. Make sure that heading to the message tabs still works and you are able to see your old messages as well as send new messages
5. Make sure that repeatedly switching tabs inside the same course will not lead to multiple `conversations` calls
6. Make sure that after receiving an unread message you see a red dot next to the message tab when reloading the page in any tab

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| course-overview.component.ts | 93% | ✅                        |
| metis-conversation.service.ts   | 86% | ✅                        |
